mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -517,69 +516,75 @@ class WidgetAdapter(
-            }
+            seekBar.addOnSliderTouchListener(this)
+            seekBar.setLabelFormatter(this)
         }
 
         override fun bind(widget: Widget) {
             super.bind(widget)
             boundWidget = widget
 
-            val stepCount = (widget.maxValue - widget.minValue) / widget.step
-            seekBar.max = Math.ceil(stepCount.toDouble()).toInt()
-            seekBar.progress = 0
+            // Fix "The stepSize must be 0, or a factor of the valueFrom-valueTo range" exception
+            seekBar.valueTo = widget.maxValue - (widget.maxValue - widget.minValue).rem(widget.step)
+            seekBar.valueFrom = widget.minValue
+            seekBar.stepSize = if (widget.step == 1F) 0F else widget.step
 
             val item = widget.item
             val state = item?.state ?: return
 
             if (item.isOfTypeOrGroupType(Item.Type.Color)) {
                 val brightness = state.asBrightness
                 if (brightness != null) {
-                    seekBar.max = 100
-                    seekBar.progress = brightness
+                    seekBar.valueFrom = 0F
+                    seekBar.valueTo = 100F
+                    seekBar.value = brightness.toFloat()
                 }
             } else {
                 val number = state.asNumber
                 if (number != null) {
-                    val progress = (number.value - widget.minValue) / widget.step
-                    seekBar.progress = Math.round(progress)
+                    var progress = number.value
+                    progress = min(progress, seekBar.valueTo)
+                    progress = max(progress, seekBar.valueFrom)
+                    seekBar.value = progress
                 }
             }
         }
 
         override fun handleRowClick() {
-            if (boundWidget?.switchSupport == true) {
-                connection.httpClient.sendItemCommand(boundWidget?.item,
-                    if (seekBar.progress == 0) "ON" else "OFF")
+            val widget = boundWidget ?: return
+            if (widget.switchSupport) {
+                connection.httpClient.sendItemCommand(widget.item,
+                    if (seekBar.value <= widget.minValue) "ON" else "OFF")
             }
         }
 
-        override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+        override fun onStartTrackingTouch(slider: Slider) {
             // no-op
         }
 
-        override fun onStartTrackingTouch(seekBar: SeekBar) {
-            Log.d(TAG, "onStartTrackingTouch position = ${seekBar.progress}")
+        override fun onStopTrackingTouch(slider: Slider) {
+            val value = slider.value
+            Log.e(TAG, "onValueChange value = $value")
+            val item = boundWidget?.item ?: return
+            if (item.isOfTypeOrGroupType(Item.Type.Color)) {
+                connection.httpClient.sendItemCommand(item, value.toString())
```

Comment:
This is now a float string, instead of the integer it was before. Can the server deal with that?

Reply:
I just tested it with a color bulb and it works fine.

Reply:
I added a function to remove trailing `.0` from floats.